specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json

```diff
@@ -1071,6 +1121,75 @@
       "required": [
         "name"
       ]
+    },
+    "workloadProfileStates": {
+      "description": "Collection of all the workload Profile States for a Premium Managed Environment..",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of workload Profile."
+        },
+        "location": {
+          "type": "string",
+          "description": "Region of the workload Profile."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of workload profile."
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of workload Profile."
+        },
```

Comment:
These are duplicates of properties from ProxyResource definition. Also if you have location, this would be a TrackedResource. Recommend removing these and changing the allOf $ref to TrackedResource from v3 common types.

Reply:
This needs to be proxy resource. Let us remove location for this.